packages/common/services/native-x.js

```diff
@@ -0,0 +1,42 @@
+const { isObject } = require('@base-cms/utils');
+const fetch = require('node-fetch');
+
+const retrieve = async ({ uri, placementId, opts }) => {
+  const query = isObject(opts) ? `?opts=${encodeURIComponent(JSON.stringify(opts))}` : '';
+  const url = `${uri}/placement/${placementId}.html${query}`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    const err = new Error(response.statusMessage);
+    err.statusCode = response.statusText;
+    throw err;
+  }
+  return response.text();
+};
+
+const getURI = site => site.get('nativeX.uri');
+
+const getPlacementId = (name, site) => site.get(`nativeX.placements.${name}.id`);
```

Comment:
Shouldn't this also get the section/primarySection to return the right placement? In existing EBM nX config we have topic-based placements, [topic] load more, [topic] hero, etc.